Update list-style and list-style-type accessibility concerns (#20009)
mdn/content: files/en-us/web/css/list-style/index.md

## Accessibility concerns

> In a notable exception, Safari will not recognize an ordered or unordered list as a list in the accessibility tree if it has a `list-style` value of `none`. This behavior is an intentional decision meant to combat misuse of list elements.

Reviewer: ```suggestion
In a notable exception, Safari will not recognize an ordered or unordered list as a list in the accessibility tree if it has a `list-style` value of `none`. This [behavior is intentional]() and not considered a bug.
```
'combat misuse of list items' is very opinionated without references. I this is by design rather than a bug, a link to the discussion is needed. I don't know the resource, so just adding `()`

Author: Great point, thank you!

The most straightforward way to address this is to add an explicit `role="list"` to the `<ol>` or `<ul>` element in the markup. This will restore the list semantics without affecting the design:

```html
<ul role="list">
  <li>An item</li>
  <li>Another item</li>
</ul>
```

> A CSS-only workaround is also available for those who do not have access to the markup: Adding any [pseudo-content](/en-US/docs/Web/CSS/content) before each list item will also restore list semantics:

Reviewer: does an empty string work? If yes, that would likely be a more appreciated code example.

```css
ul {
  list-style: none;
}

ul li::before {
  content: "+ ";
}
```

> If the intent is to keep list item markers visually hidden, this can be managed with a [zero-width space](https://en.wikipedia.org/wiki/Zero-width_space):

Reviewer: It's in the CSS, so it's fine, but maybe add 
> `&#8203;`, which is '\200B' in CSS and '\u200B' in JavaScript.
totally optional edit.

Author: I can see how that would improve clarity here. I'll add it in. Thank you!
